Dockerfile

```diff
@@ -48,6 +59,59 @@ COPY dist/$TARGETOS/$TARGETARCH/vault-k8s /bin/
 USER vault
 ENTRYPOINT ["/bin/vault-k8s"]
 
+# This target creates a production ubi release image
+# for the project for use on OpenShift.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6 as ubi
+
+ARG PRODUCT_NAME
+ARG PRODUCT_VERSION
+ARG BIN_NAME
+
+# PRODUCT_NAME and PRODUCT_VERSION are the name of the software on releases.hashicorp.com
+# and the version to download. Example: PRODUCT_NAME=vault-k8s PRODUCT_VERSION=1.0.0.
+ENV BIN_NAME=$BIN_NAME
+ENV PRODUCT_VERSION=$PRODUCT_VERSION
+
+ARG PRODUCT_NAME=$BIN_NAME
+
+# TARGETOS and TARGETARCH are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+# Set ARGs as ENV so that they can be used in ENTRYPOINT/CMD
+ENV PRODUCT_VERSION=$PRODUCT_VERSION
+ENV BIN_NAME=$BIN_NAME
```

Comment:
It looks like these were already defined on lines 72-73?

Reply:
Oh good catch, ty!